incydr/utils.py

```diff
@@ -54,15 +54,41 @@ def read_models_from_csv(
     reader = DictReader(file)
 
     error = None
-    for i, row in enumerate(reader, 1):
+    for row in reader:
         for key, val in row.items():
             if val == "":
                 row[key] = None
         try:
             yield model(**row)  # noqa
         except ValidationError as err:
             error = CSVValidationError(
-                f"Bad data in row {i} of {path}\n{str(err)}", row=i
+                f"Bad data in row {reader.line_num} of {path}\n{str(err)}",
+                row=reader.line_num,
+            )
+    if error:
+        raise error
+
+
+def read_dict_from_csv(
+    path: Union[str, Path, IOBase], field_names: List[str] = None
+) -> Generator[dict, None, None]:
+    if isinstance(path, IOBase):
+        file = path
+    else:
+        path = Path(path)
+        file = path.open(mode="r", encoding="utf-8")
+    reader = DictReader(file, fieldnames=field_names)
+    error = None
+    for row in reader:
+        for key, val in row.items():
+            if val == "":
+                row[key] = None
+        try:
+            yield row  # noqa
+        except ValidationError as err:
```

Comment:
`ValidationError` won't ever get raised here since pydantic isn't doing anything in this function, right? What's the use-case for this in addition to the `read_model_from_csv()`?

Reply:
Ah, no you're right I just copy pasted and didn't double check where that error was coming from.  I think I assumed it was from the `csv` package, I'll adjust it.